app/src/main/java/com/mapbox/navigation/examples/basics/ShowCameraTransitionsActivity.kt

```diff
@@ -36,48 +88,317 @@ class ShowCameraTransitionsActivity : AppCompatActivity() {
         } else {
             MapboxNavigationProvider.create(
-                NavigationOptions.Builder(this)
+                NavigationOptions.Builder(this.applicationContext)
                     .accessToken(getString(R.string.mapbox_access_token))
                     // comment out the location engine setting block to disable simulation
                     .locationEngine(replayLocationEngine)
                     .build()
             )
         }
+
+        // initialize Navigation Camera
+        viewportDataSource = MapboxNavigationViewportDataSource(mapboxMap)
+        navigationCamera = NavigationCamera(
+            mapboxMap,
+            binding.mapView.camera,
+            viewportDataSource
+        )
+        // set the animations lifecycle listener to ensure the NavigationCamera stops
+        // automatically following the user location when the map is interacted with
+        binding.mapView.camera.addCameraAnimationsLifecycleListener(
+            NavigationBasicGesturesHandler(navigationCamera)
+        )
+        navigationCamera.registerNavigationCameraStateChangeObserver { navigationCameraState ->
+            // shows/hide the recenter button depending on the camera state
+            when (navigationCameraState) {
+                NavigationCameraState.TRANSITION_TO_FOLLOWING,
+                NavigationCameraState.FOLLOWING ->
+                    binding.recenterButton.visibility = View.INVISIBLE
+                NavigationCameraState.TRANSITION_TO_OVERVIEW,
+                NavigationCameraState.OVERVIEW,
+                NavigationCameraState.IDLE -> binding.recenterButton.visibility = View.VISIBLE
+            }
+        }
+        // set the padding values depending to correctly frame maneuvers and the puck
+        viewportDataSource.overviewPadding = overviewPadding
+        viewportDataSource.followingPadding = followingPadding
+
+        // initialize route line, the withRouteLineBelowLayerId is specified to place
+        // the route line below road labels layer on the map
+        // the value of this option will depend on the style that you are using
+        // and under which layer the route line should be placed on the map layers stack
+        val mapboxRouteLineOptions = MapboxRouteLineOptions.Builder(this)
+            .withRouteLineBelowLayerId("road-label")
+            .build()
+        routeLineApi = MapboxRouteLineApi(mapboxRouteLineOptions)
+        routeLineView = MapboxRouteLineView(mapboxRouteLineOptions)
+
+        // add click listeners for buttons
+        binding.recenterButton.setOnClickListener {
+            navigationCamera.requestNavigationCameraToFollowing()
+        }
+        binding.overviewButton.setOnClickListener {
+            navigationCamera.requestNavigationCameraToOverview()
+        }
+        binding.simulationButton.setOnClickListener {
+            // restarts the simulation
+            startSimulation(route)
+        }
```

Comment:
Do we really need `simulationButton`? All this button does is call `startSimulation(route)`. I think we can safely remove it, as there is no additional API's that are invoked on the `NavigationCamera` since it just updates the camera position based on `routeProgress` and in this case restarts the puck position from the beginning of the route.

Reply:
Removed :+1: